test/client/client_spec.rb

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2016 SUSE LLC.
```

Comment:
Well, it should if you change the implementation as suggested above (with a separate class in `/lib`).